tfexec/terraform_state_replace_provider.go

```diff
@@ -0,0 +1,71 @@
+package tfexec
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/hashicorp/go-version"
+)
+
+const (
+	// MinimumTerraformVersionForStateReplaceProvider specifies the minimum
+	// supported Terraform version for StateReplaceProvider.
+	MinimumTerraformVersionForStateReplaceProvider = "0.13"
+)
+
+// StateReplaceProvider replaces providers from source to destination address.
+// If a state argument is given, use it for the input state.
+// It returns the given state.
+func (c *terraformCLI) StateReplaceProvider(ctx context.Context, state *State, source string, destination string, opts ...string) (*State, error) {
+	constraints, err := version.NewConstraint(fmt.Sprintf(">= %s", MinimumTerraformVersionForStateReplaceProvider))
+	if err != nil {
+		return nil, err
+	}
+
+	v, err := c.Version(ctx)
```

Comment:
It may be ideal for getting and caching the version at the time of initialization of terraformCLI. Still, the scope of refactoring would be significant, so we will postpone it for now.